edu/wisc/ssec/mcidasv/resources/python/background.py

```diff
@@ -496,6 +518,33 @@ def getDefaultLayerLabel(self):
         return defaultLabel
 
 
+def _dictFromAttributeList(attrList):
+    """Build a attrName->attrValue dictionary from a list of NetCDF Attributes."""
+    result = dict()
+    for attr in attrList:
+        result[attr.getFullName()] = _getNetcdfAttributeValue(attr)
+    return result
+
+
+def _getNetcdfAttributeValue(attr):
+    """Helper to return netcdf Attribute value as either numeric or string value."""
+    if attr.isArray():
+        # umm. convert to python list i guess?
+        values = []
+        array = attr.getValues()
+        if array.getRank() != 1:
+            raise ValueError("Why would an attribute be multi-dimensional???")
```

Comment:
Just for the sake of completeness: https://github.com/Unidata/thredds/blob/master/cdm/src/main/java/ucar/nc2/Attribute.java#L47-L48

> An Attribute has a name and a value, used for associating arbitrary metadata with a Variable or a Group.
> The value can be a one dimensional array of Strings or numeric values.